### PR TITLE
[Merged by Bors] - feat: port Algebra.Category.Mon.FilteredColimits

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Category/Mon/FilteredColimits.lean [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+noncomputable local instance FunLike_instance (t : Cocone.{v, max v u, v} F) :
+  FunLike (F.obj j ⟶ t.pt)
+    ((F ⋙ forget CommMonCat).obj j)
+    (fun _ => t.pt) :=
+show FunLike (F.obj j →* t.pt) ((F ⋙ forget CommMonCat.{max v u}).obj j)
+  (fun _ => t.pt) by infer_instance
+
+-- Porting note : need to add `FunLike` instance manually
+@[to_additive]
+noncomputable local instance FunLike_instance' (t : Cocone.{v, max v u, v} F) :
```
Comment (ChrisHughes24) [inline Mathlib/Algebra/Category/Mon/FilteredColimits.lean:460]: Why are these only local instances? What is going on here, why can't they be synthesised?

Comment (jjaassoonn) [reply Mathlib/Algebra/Category/Mon/FilteredColimits.lean:460]: I suspect Lean don't know the arrows in category of monoids are just monoid homomorphisms. Lean knows that homomorphisms are `FunLike` but can't see that arrows are FunLike unless it is expanded manually.

Comment (jjaassoonn) [reply Mathlib/Algebra/Category/Mon/FilteredColimits.lean]: I stated `FunLike` instance for arrows in `(Comm)MonCat`in `MonCat` file, so local instances no longer needed

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**